lms/djangoapps/instructor/views/api.py

```diff
@@ -772,6 +772,49 @@ def extract_user_info(user):
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @require_level('staff')
+def get_problem_responses(request, course_id):
+    """
+    Initiate generation of a CSV file containing all student answers
+    to a given problem.
+
+    Responds with JSON
+        {"status": "... status message ..."}
+
+    if initiation is successful (or generation task is already running).
+
+    Responds with BadRequest if problem URL is faulty.
+    """
+    course_key = CourseKey.from_string(course_id)
+    problem_url = request.GET.get('problem_url', '')
+
+    if problem_url[-4:] == ".xml":
+        problem_url = problem_url[:-4]
+
+    try:
+        course_key.make_usage_key_from_deprecated_string(problem_url)
+    except InvalidKeyError:
+        problem_not_found_status = _("Could not find problem with this URL.")
+        return JsonResponseBadRequest(problem_not_found_status)
```

Comment:
@bradenmacdonald Also not 100% sure what formats need to be accepted but my guess is that we'll probably need to support the `i4x://OpenCraft/ID101/problem/3ede77cb1bd64138850a6cad74e461ff` format (even though it's deprecated it still seems to be used all over the place).

I gave the code above a try. It almost works, except for the fact that `problem_key.course` doesn't match `course_key` (at least for URLs in the `i4x` format):

    COURSE_KEY: OpenCraft/ID101/2015
    PROBLEM_KEY: i4x://OpenCraft/ID101/problem/3ede77cb1bd64138850a6cad74e461ff
    PROBLEM_KEY.COURSE: ID101

So when I modify it to compare `problem_key.course` and `course_key.course`, it won't raise `InvalidKeyError`. However, I'm not sure if it's enough to compare the `course` attributes of the two keys? As in: Would it be better to ensure that `org` (`OpenCraft`) and `run` (`2015`) also match?